### PR TITLE
various: better UTC times

### PR DESCRIPTION
**Pull request by rerowep**
Co-Authored-by: Peter Weber <peter.weber@rero.ch>

### CODE REVIEW

File: rero_mef/agents/mef/api.py [hunk 1]
```diff
@@ -166,6 +168,20 @@ def create_or_update_mef_viaf_record(self, dbcommit=False, reindex=False,
...
+        """Create a deleted record for an record.
+
+        :param record: Record to create.
+        :param dbcommit: Commit changes to DB.
+        :param reindex: Reindex record.
+        :returns: Created record.
+        """
+        data = {record.name: {
+            '$ref': build_ref_string(record.pid, record.name)}}
+        data['deleted'] = datetime.now(timezone.utc).isoformat()
```
Comment (zannkukai) [inline rero_mef/agents/mef/api.py:182]: could be merge to increase readability (my opinion)
```python
data = {
    record.name: {
        '$ref': build_ref_string(record.pid, record.name)
    },
    deleted: datetime.now(timezone.utc).isoformat()
}
```
This method could be placed into the parent `EntityMefRecord` to declare it once and use them in all children classes.

File: rero_mef/concepts/mef/api.py [hunk 2]
```diff
@@ -112,6 +114,20 @@ def create_or_update_mef_viaf_record(self, dbcommit=False, reindex=False,
...
+        :param record: Record to create.
+        :param dbcommit: Commit changes to DB.
+        :param reindex: Reindex record.
+        :returns: Created record.
+        """
+        data = {record.name: {
+            '$ref': build_ref_string(record.pid, record.name)}}
+        data['deleted'] = datetime.now(timezone.utc).isoformat()
+        return cls.create(data=data, dbcommit=dbcommit, reindex=reindex)
+
```
Comment (zannkukai) [inline rero_mef/concepts/mef/api.py:130]: if places into parent `EntityMefRecord`, you can remove due to inheritance.

File: rero_mef/utils.py [hunk 3]
```diff
@@ -387,30 +392,31 @@ def oai_save_records_from_dates(name, file_name, sickle, oai_item_iterator,
             }
             if access_token:
                 params['accessToken'] = access_token
-            params.update(dates)
+            params |= dates
```
Comment (zannkukai) [inline rero_mef/utils.py:395]: This operation on dict are only available with `>=python3.9`. If you want MEF project stays compatible with older project, continue to use `.update()` method.
We had the same problem with `rero-invenio-base` project, we choose to keep old method to not force children project to have python3.9.


(Sourcery plugin proposition I think ;-) )

File: rero_mef/utils.py [hunk 4]
```diff
@@ -387,30 +392,31 @@ def oai_save_records_from_dates(name, file_name, sickle, oai_item_iterator,
...
                 if until_date > my_until_date:
                     until_date = my_until_date
                 dates = {
-                    'from': my_from_date.strftime("%Y-%m-%d"),
-                    'until': until_date.strftime("%Y-%m-%d")
+                    'from': my_from_date.strftime(TIME_FORMAT),
+                    'until': until_date.strftime(TIME_FORMAT)
                 }
-                params.update(dates)
+                params |= dates
```
Comment (zannkukai) [inline rero_mef/utils.py:412]: same here

**Review by zannkukai — CHANGES_REQUESTED** [review]

File: rero_mef/concepts/mef/api.py (hunk 2 above)
Comment (rerowep) [reply rero_mef/concepts/mef/api.py:130]: I have to rethink the Classes. Normally I have to create a entity and a MEF class.

File: rero_mef/utils.py (hunk 3 above)
Comment (rerowep) [reply rero_mef/utils.py:395]: No problem here MEF is python version 3.9

**Review by zannkukai — APPROVED** [review]

**Pull request merged**